### PR TITLE
bpo-33921: Clarify how to bind to all interfaces using socket

### PR DESCRIPTION
**Pull request by johnthagen**
Fixes https://bugs.python.org/issue33921

https://bugs.python.org/issue33921

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


When your account is ready, please add a comment in this pull request
and a Python core developer will remove the `CLA not signed` label
to make the bot check again.

Thanks again for your contribution, we look forward to reviewing it!

**Comment by johnthagen** [comment]
Updated b.p.o account to associate GitHub username.

File: Doc/library/socket.rst [hunk 1]
```diff
@@ -69,6 +69,13 @@ created.  Socket addresses are represented as follows:
   where *host* is a string representing either a hostname in Internet domain
   notation like ``'daring.cwi.nl'`` or an IPv4 address like ``'100.50.200.5'``,
   and *port* is an integer.
+  
+  - For IPv4 addresses, two special forms are accepted instead of a host
```
Comment (ammaraskar) [inline Doc/library/socket.rst]: You've got trailing whitespace on some of these lines, aside from that the content looks good.

**Review by ammaraskar — CHANGES_REQUESTED** [review]

**Comment by ammaraskar** [comment]
Still one line left with trailing whitespace: https://travis-ci.org/python/cpython/jobs/395902432#L544

fyi, you can catch these locally before pushing by running `make patchcheck` https://devguide.python.org/pullrequest/#patchcheck

**Comment by johnthagen** [comment]
@ammaraskar  Thanks for the heads up. Been editing in the browser for this for simplicity, but I'll note that for next time if it's a more substantial PR.

**Review by ammaraskar — APPROVED** [review]
Looks good to me, can be marked with skip news since its a minor doc change.

**Comment by johnthagen** [comment]
@ned-deily Are you available to review?

**Review by tiran — APPROVED** [review]

**Pull request merged**